ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -776,6 +776,18 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 			return err
 		}
 
+		// Delete all workers if worker group is suspended and skip reconcile
+		if worker.Suspend != nil && *worker.Suspend {
+			if _, err := r.deleteAllPods(ctx, common.RayClusterGroupPodsAssociationOptions(instance, worker.GroupName)); err != nil {
+				r.Recorder.Eventf(instance, corev1.EventTypeWarning, string(utils.FailedToDeleteWorkerPod),
+					"Failed deleting worker Pods for suspended group %s in RayCluster %s/%s, %v", worker.GroupName, instance.Namespace, instance.Name, err)
+				return errstd.Join(utils.ErrFailedDeleteWorkerPod, err)
+			}
```

Comment:
Should we make the suspend operation atomic, like in https://github.com/ray-project/kuberay/pull/1798? 

If this field is only used by the RayJob controller for now, it is fine for it not to be atomic at this moment. In that case, we should add a comment in `raycluster_types.go` to state that this field is not user-facing and add a check in `validateRayClusterSpec` to ensure that `workerGroup.suspend` is not `false`—only `nil` and `true` are valid.

Does `DeleteAllOf` guarantee to delete all or nothing? Is it possible that only some Pods are deleted while others fail to be deleted? If `DeleteAllOf` guarantee to delete all or nothing, we don't need to worry about the atomicity.

Reply:
I don't think DeleteAllOf is atomic. Not sure if the same approach will work for RayJob, this means we'll need a condition per worker group to track whether the suspension is completed

Reply:
Are we expecting this field to be only for the RayJob controller and not user-facing, at least for v1.3.0? If so, we can use:

> If this field is only used by the RayJob controller for now, it is fine for it not to be atomic at this moment. In that case, we should add a comment in raycluster_types.go to state that this field is not user-facing and add a check in validateRayClusterSpec to ensure that workerGroup.suspend is not false—only nil and true are valid. This is not ideal, but may be able to avoid most misconfig.

Reply:
I think the main use-case will be with RayJob, but I think it should be valid for other use-cases. We shouldn't stop or prevent users from using this field if they want to